install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -93,13 +93,62 @@ spec:
-    - name: master-nodes-high-memory-usage
+    - name: high-overall-control-plane-memory
       rules:
-        - alert: MasterNodesHighMemoryUsage
+        - alert: HighOverallControlPlaneMemory
           expr: |
-            ((sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) - sum(node_memory_MemFree_bytes + node_memory_Buffers_bytes + node_memory_Cached_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" ))) / sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) * 100) > 90
-          for: 15m
+            (
+              1
+              -
+              sum(
+                node_memory_MemFree_bytes
+                + node_memory_Buffers_bytes
+                + node_memory_Cached_bytes
+                AND on (instance)
+                label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )
+              ) / sum(
+                node_memory_MemTotal_bytes
+                AND on (instance)
+                label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )
+              )
+            ) * 100 > 60
+          for: 1h
           labels:
             severity: warning
           annotations:
-            message: "Memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90%. Master nodes starved of memory could result in degraded performance of the control plane."
+            summary: >-
+              Memory utilization across all control plane nodes is high, and could impact responsiveness and stability.
+            description: >-
+              Given three control plane nodes, the overall memory utilization may only be about 2/3 of all available capacity.
+              This is because if a single control plane node fails, the kube-apiserver and etcd my be slow to respond.
+              To fix this, increase the memory of the control plane nodes.
+    - name: extremely-high-control-plane-memory
+      rules:
+        - alert: ExtremelyHighControlPlaneMemory
+          expr: |
+            (
+              1
+              -
+              sum(
+                node_memory_MemFree_bytes
+                + node_memory_Buffers_bytes
+                + node_memory_Cached_bytes
+                AND on (instance)
+                label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )
+              ) / sum(
+                node_memory_MemTotal_bytes
+                AND on (instance)
+                label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )
+              )
+            ) * 100 > 90
+          for: 30m
+          labels:
+            severity: critical
```

Comment:
This should not be critical IMO - these conditions do not illustrate any service degredation.

Reply:
Over 90% for an hour is worth a midnight page for me.  But if folks suspect this will be going off all the time (😱), I don't mind opening with `warning` and revisiting in 4.12 once we have more data.

Reply:
I agree with @wking, this is definitely worth a page. A 90% CPU usage in the control plane, is very likely to result in a degradation of your overall cluster because the apiserver, etcd, ... won't be able to handle the load.

It is worth noting that since `critical` was kept, you need to add a runbook to the alert.